### ISSUE TITLE
Application-wide component defaults

### ISSUE THREAD

**Issue by Cosmatevs** [radzenhq/radzen-blazor#967]
**Is your feature request related to a problem? Please describe.**
It's quite annoying when I want to use consistently e.g. 600px wide draggable dialogs, filled form fields, and data grid pagers on top, and I need to declare it each time I create instances of those components. It becomes even worse when I have to change such a setting and find all the places where it was used.

**Describe the solution you'd like**
Default values declared in Program.cs/Startup.cs or in appsettings.json.

**Describe alternatives you've considered**
None.

**Additional context**
[HAVIT Blazor has a static _Defaults_ property in some component classes.](https://havit.blazor.eu/concepts/defaults-and-settings)

**Comment by akorchev** [comment]
Making a static class and changing all components to use it would be a humongous effort that we prefer not to take yet. Static classes also introduce an issue with certain aspects of an application (for example localization - the same app won't be able to server two localized versions if it is using a single static class).

Wouldn't using custom (wrapped) components with preset default values serve a similar purpose? Perhaps inheritance would be even easier:

```
public class MyDataGrid<TItem> : RadzenDataGrid<TItem>
{
      public MyDataGrid()
      {
         PagerPosition = PagerPosition.Top;
      }
}
```

**Comment by Cosmatevs** [comment]
I tried to create a wrapper but it was very troublesome to bind values in it. I haven't thought about inheritance and haven't even known it's so easy to inherit whole component with its _.razor_ part. Thank you for sharing that solution 🙂
I'm not very convinced though to making new classes for each component I want to have with custom default values. Me and others who develop the same applications would have to remember to use them instead of Radzen components. And remember to create new classes if needed and replace usages to stay consistent 😅

HAVIT handles defaults with [a static _Defaults_ property in each component class](https://github.com/havit/Havit.Blazor/blob/989338da09bc29f7486f595b094db6a2c9a9afc2/Havit.Blazor.Components.Web.Bootstrap/Buttons/HxButton.razor.cs#LL16C1-L29C3) of a specific settings type (e.g. _ButtonSettings_), not with a static class. Also not all properties can have custom default values.
Note that I only gave HAVIT as an example. Defaults could be implemented any other way 😀 However, for now, I don't have a specific idea how to implement it.
